src/lists/DoublyLinkedIntList.java

```diff
@@ -0,0 +1,383 @@
+
+        if (index < 0 || index >= size) {
+                throw new IndexOutOfBoundsException(
+                    "Index must be in the Range 0-" + (size - 1));
+        } else {
+            //Skip looping if index is at front of list
+            if (index == 0) {
+                return current.data;
+                // Skip looping if index is at back of list
+            } else if (index == size - 1) {
+                return back.prev.data;
+                // If index is in acceptable range loop to find Node at index value
+            } else {
+                while (current != back) {
+                    if (index == dex) {
+                        return current.data;
+                    } else {
+                        dex++;
+                        current = current.next;
+                    }
+                } return current.data;
+            }
+        }
+    }
+
+    /**
+     * Returns true if this list contains the specified value.
+     *
+     * @param value value whose presence in this list is to be searched for
+     * @return true if this list contains the specified value
+     */
+    @Override
+    public boolean contains(int value) {
+
+        Node current = front.next;
+
+        while (current != back) {
+            if (current.data == value) {
+                return true;
+            } else {
+                current = current.next;
+            }
+        } return false;
+    }
+
+    /**
+     * Returns the index of the first occurrence of the specified value
+     * in this list, or -1 if this list does not contain the value.
+     *
+     * @param value value to search for
+     * @return the index of the first occurrence of the specified value in this list
+     * or -1 if this list does not contain the value
+     */
+    @Override
+    public int indexOf(int value) {
+
+        int dex = 0;
+        Node current = front.next;
+
+        if (contains(value)) {
+             while (current.data != value) {
+                 dex++;
+                 current = current.next;
+             } return dex;
+        } return -1;
+    }
+
+    /**
+     * Returns true if this list contains no values.
+     * @return true if this list contains no values
+     */
+    @Override
+    public boolean isEmpty() {
+        return front.next == back && back.prev == front;
+    }
+
+    /**
+     * Returns the number of values in this list.
+     * @return the number of values in this list
+     */
+    @Override
+    public int size() {
+        return this.size;
+    }
+
+    /**
+     * Removes all the values from this list.
+     * The list will be empty after this call returns.
+     */
+    @Override
+    public void clear() {
+        front.next = back;
+        back.prev = front;
+        size = 0;
+    }
+
+    /**
+     * Returns an iterator over elements of type {@code T}.
+     * @return an Iterator.
+     */
+    @Override
+    public Iterator<Integer> iterator() {
+        return null;
```

Comment:
I see your iterator() method on the DoublyLinkedList class returns null. I believe that is a stub. It should return an instance of a new iterator object. If you look at the LinkedIntList (singly-linked list), you will see that you defined an iterator class (nested) that has implementations of next() and hasNext() methods -- and then in the iterator() method returned a new instance of that class. We need the same/similar here in DoublyLinkedList...